pyproject.toml

```diff
@@ -87,60 +106,19 @@ ignore_missing_imports = true
 [tool.pytest.ini_options]
 junit_family = "xunit2"
 
-[tool.pdm.dev-dependencies]
-dev = [
-    "pytest",
-    "pytest-mock",
-    "mypy",
-    "taskipy",
-    "pytest-cov",
-    "python-multipart",
-    "types-PyYAML<7.0.0,>=6.0.3",
-    "types-certifi<2021.10.9,>=2020.0.0",
-    "types-python-dateutil<3.0.0,>=2.0.0",
-    "ruamel-yaml-string>=0.1.1",
-    "syrupy>=4",
-]
-
-[tool.pdm.build]
-includes = [
-    "openapi_python_client",
-    "CHANGELOG.md",
-    "openapi_python_client/py.typed",
-]
-
-[tool.pdm.scripts]
+[tool.taskipy.tasks]
 lint = "ruff check --fix ."
 format = "ruff format ."
-mypy = "mypy openapi_python_client"
-check = { composite = ["lint", "format", "mypy", "test"] }
-regen = {composite = ["regen_e2e", "regen_integration"]}
+mypy = "mypy openapi_python_client"  # note, currently test code is excluded from type checking
+check = "task lint && task format && task mypy && task test"
+regen = "task regen_e2e && task regen_integration"
 e2e = "pytest openapi_python_client end_to_end_tests/test_end_to_end.py"
-re = {composite = ["regen_e2e", "e2e --snapshot-update"]}
+re = "task regen_e2e && task e2e --snapshot-update"
 regen_e2e = "python -m end_to_end_tests.regen_golden_record"
-
-[tool.pdm.scripts.test]
-cmd = "pytest tests end_to_end_tests/test_end_to_end.py end_to_end_tests/functional_tests --basetemp=tests/tmp"
-[tool.pdm.scripts.test.env]
-"TEST_RELATIVE" = "true"
```

Comment:
This is the reason I added `python-dotenv` as a dev dependency. There are some extra unit tests that only get run in CI which are enabled by setting this env var. However, we're now using `taskipy` instead of pdm tasks, and `taskipy` unfortunately has no built-in way to set env vars. So I added a `.env` file in the tests and added `dotenv run` in the test_with_coverage task below.